nixos/modules/services/networking/seafile.nix

```diff
@@ -0,0 +1,272 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  python = pkgs.python3Packages.python;
+  cfg = config.services.seafile;
+
+  ccnetConf = pkgs.writeText "ccnet.conf" ''
+    [General]
+    SERVICE_URL = ${cfg.serviceUrl}
+  '';
+
+  seafileConf = pkgs.writeText "seafile.conf" ''
+    [fileserver]
+    port=${toString cfg.fileserverPort}
+    host=${cfg.fileserverHost}
+    ${cfg.seafileExtraConf}
+  '';
+
+  seahubSettings = pkgs.writeText "seahub_settings.py" ''
+    FILE_SERVER_ROOT = '${cfg.serviceUrl}/seafhttp'
+    DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.sqlite3',
+            'NAME': '${seahubDir}/seahub.db',
+        }
+    }
+    MEDIA_ROOT = '${seahubDir}/media/'
+    THUMBNAIL_ROOT = '${seahubDir}/thumbnail/'
+
+    with open('${cfg.seahubSecretKeyFile}') as f:
+        SECRET_KEY = f.readline().rstrip()
+
+    ${cfg.seahubExtraConf}
+  '';
+
+  seafRoot = "/var/lib/seafile"; # hardcode it due to dynamicuser
```

Comment:
We probably shouldn't use a dynamic user here due to possible permission issues.

Reply:
Other than the hardcoded path I don't see any downside? DynamicUser is rather convenient here. What permission issues do you have in mind?

Reply:
@SuperSandro2000 did you have some specific permissions issues in mind? I think I get what you're saying... but if `DynamicUser` works maybe we should ship with that.

Reply:
Oh also something something something [JoinsNamespaceOf](https://www.freedesktop.org/software/systemd/man/systemd.unit.html#JoinsNamespaceOf=) something something... maybe :man_shrugging:

Reply:
> did you have some specific permissions issues in mind?

The files it places there could have permission issues.

Reply:
The combination of `DynamicUser`, `User`, `Group`, and `StateDirectory` should take care of permissions appropriately.

Reply:
Can someone tell me why `DynamicUser` requires the directory to be hardcoded? I consider the state directory location root to be a rather important configuration parameter for an installation.
I myself prefer `/srv/seafile` as a location – at least for data – which is more in line with FHS conventions – which NixOS does not care about – than with Debian conventions – which NixOS shouldn't care about either.

Reply:
You  would have to discuss that with the `systemd` folks... this has nothing to do with NixOS and isn't in any way under our control. If I had to imagine I might guess it has to do with decades of convention and `/var`...